sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,6 +114,17 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
+      if (actualSchema.isEmpty && requiredSchema.length == 1 &&
```

Comment:
Actually we don't need the first condition`actualSchema.isEmpty`. The last two already guarantee it.

Reply:
Also actually we should move this check out of this function block.